contrib/kubespray/roles/port/defaults/main.yml

```diff
@@ -0,0 +1,2 @@
+host_daemon_port_start: 1024
+host_daemon_port_end: 19999
```

Comment:
I found this is not a good range, for example, job exporter uses 9102 port on worker nodes, if kubelet occupies this port during k8s deployment, job exporter won't be started anymore.

It's better to avoid port conflicts with system/pai services, that's why the default value starts from 32768.

Reply:
Let's reduce the ports range of non-job ports, so job ports range can be larger and less possible to conflict

Reply:
May be we can tune this to be 10000-19999?

Reply:
another choice is using `net.ipv4.ip_local_reserved_ports` instead, you can refer to [IP Variables](https://www.kernel.org/doc/Documentation/networking/ip-sysctl.txt) for details. you can also reuse the job port range for this config.

---

when calling `connect()` or `bind()` with port number 0, OS will choose port which which meets all of the following:
1. in `net.ipv4.ip_local_port_range`
2. not in `net.ipv4.ip_local_reserved_ports`
3. not **currently** in use

Reply:
But each port need to specify in ip_local_reserved_ports, and we have too many services. A range is easy to maintain?

Reply:
1. let `net.ipv4.ip_local_port_range` uses default range 32768~60999
2. we set job port range in `net.ipv4.ip_local_reserved_ports`, e.g. 20000~40000
3. services use static ports, no need to reserve, kubelet will be assigned a free port between 40000~60999 when binding to port 0

Reply:
What I meant is ip_local_reserved_ports is not a range.

Reply:
How about just set `net.ipv4.ip_local_port_range` to 40000~60000. And we assume:
PAI-service use port within 1024-19999.
PAI-job use port within 20000-39999
Random port is: 40000-60000

For PAI service, we need to make sure the port number doesn't conflict with system service.
Random port and PAI job port will not conflict according to this config.

Reply:
Agree. But lets enlarge PAI-job ports range, such as 
PAI-service use port within 1024-**9999** (given we only have very few services)
PAI-job use port within 10000-39999
Random port is: 40000-**65535**

Reply:
@abuccts just saw ip_local_reserved_ports  can also specify range, like 1-39999

Reply:
> What I meant is ip_local_reserved_ports is not a range.

it is a *list of comma separated ranges*, so it can be one range, or comma separated multiple ranges

> How about just set net.ipv4.ip_local_port_range to 40000~60000

it's OK

> PAI-job use port within 10000-39999, Random port is: 40000-65535

kubelet also uses static ports 10249 and 10250
to enlarge job ports, it's better to use 20000~50000

Reply:
Random port will also impacts jobs as they may use.
So lets make Random port and PAI-job port both large.
If you do not want to change kubelet ports, we can use range:
PAI-service use port within 1024-15000 (given we only have very few services)
PAI-job use port within 15000-39999
Random port is: 40000-65535

Reply:
Fine for me, will change random port to `40000-65535`

Reply:
seems fine